### ISSUE TITLE
Need more info on how to structure application state

### ISSUE THREAD

**Issue by rads** [omcljs/om#20]
I've been trying out Om with a UI for a music player. I've had a hard time understanding where certain pieces of state should go. My app state looks like this:

```
{:search-results [{:id 1, :title "First Song", ...}
                  {:id 2, :title "Second Song", ...}]
 :player {:playing? true
          :current-song {:id 1, :title "First Song", ...}}}
```

The `:search-results` key represents to the songs that show up in the UI. The user can click on a song to change the current song being played, which is when the `:player` key gets updated.

The confusing part is that the `:player` key does not map to a single UI component, since there are multiple parts of the UI that need to change when a new song starts playing. For example, when the `:playing?` key changes, the player's controls need to change a button to show "Play" or "Pause". At the same time, the search results need to show an icon next to the current song in the results to show the song that is currently selected.

There is a single `:player` key for both the search results and the controls components. An alternative would be to create a `:controls` key, get rid of the `:player` key, and then duplicate the player state for each component. This would match up better with the idea of representing UI components as EDN, but it seems like a bad idea to me to duplicate the application state.

It seems like with [issue #19](https://github.com/swannodette/om/issues/19) there will be another place to put the state, but I'm not sure how that would help. In plain React, there is one place to put state: encapsulated inside an component. There is no global app state, which means components can only share information using props, and only from parent to child. There's no decision to be made about whether the state is global or transient.

In summary:
- How do you deal with shared state among multiple components?
- What is the need for having both a global application state and individual component states?
- If a component has its own state, but needs to react to shared state as well, how do you express that when we are currently only allowed to set one path per component?

Thanks for the great work!
